test/parallel/test-http-write-head.js

```diff
@@ -76,3 +75,24 @@ function runTest() {
     response.resume();
   }));
 }
+
+{
+  const server = http.createServer(common.mustCall((req, res) => {
+    res.writeHead(200, [ 'test', '1' ]);
+    assert.throws(() => res.writeHead(200, [ 'test2', '2' ]), {
+      code: 'ERR_HTTP_HEADERS_SENT',
+      name: 'Error',
+    });
+    res.end();
+  }));
+
+  server.listen(0, common.mustCall(() => {
+    http.get({ port: server.address().port }, (res) => {
+      assert.strictEqual(res.headers.test, '1');
+      assert.strictEqual('test2' in res.headers, false);
```

Comment:
Nit, feel free to ignore.

```suggestion
      assert.strictEqual(res.headers.test2, undefined);
```